### PR TITLE
zareen js week-2

### PR DESCRIPTION
**Pull request by khan-zareen**

### CODE REVIEW

File: week-2/M-array-map/exercise.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ function double(num) {
 }
 
 var numbers = [1, 2, 3, 4];
-var numbersDoubled; // complete this statement (use map and the double function)
+var numbersDoubled = numbers.map(double); // complete this statement (use map and the double function)
```
Comment (martynamaron) [inline week-2/M-array-map/exercise.js]: Well done!

File: week-2/L-array-methods-2/exercise3.js [hunk 2]
```diff
@@ -6,8 +6,8 @@
 
 var ukNations = ["Scotland", "Wales", "England", "Northern Ireland"];
 
-function isInUK(country) {
-  return; // complete this statement
+function isInUk(country) {
+  return ukNations.includes(country); // complete this statement
```
Comment (martynamaron) [inline week-2/L-array-methods-2/exercise3.js]: That's right :)

File: week-2/L-array-methods-2/exercise.js [hunk 3]
```diff
@@ -15,8 +15,8 @@ var everyone = [
   "Swathi"
 ];
 
-var firstFive; // complete this statement
-var lastFive; // complete this statement
+var firstFive = everyone.slice(0, 5); // complete this statement
```
Comment (martynamaron) [inline week-2/L-array-methods-2/exercise.js]: That's correct, good understanding of array indexes 👍

File: week-2/I-array-properties/exercise.js [hunk 4]
```diff
@@ -6,9 +6,12 @@
 */
 
 function isEmpty(arr) {
-  return; // complete this statement
+  if (arr.length == 0) {
```
Comment (martynamaron) [inline week-2/I-array-properties/exercise.js]: How can you simplify this code? `isEmpty` is meant to return a boolean, which it does at the moment, but `arr.length == 0` on its own also returns a boolean! How can you achieve the same without `if` and `else`?

File: week-2/G-conditionals-2/exercise-3.js [hunk 5]
```diff
@@ -9,7 +9,15 @@
 */
 
 function calculateGrade(mark) {
-
+  if (mark >= 80) {
+    return "A";
+  } else if (mark >= 60 && mark <= 80) {
```
Comment (martynamaron) [inline week-2/G-conditionals-2/exercise-3.js]: Ideally, you will want to make sure that certain input will satisfy only one of your conditions. If the mark is `80`, both `mark >= 80` and `mark <= 80` return true. How can you fix that? the same applies for line 16

File: week-2/G-conditionals-2/exercise-2.js [hunk 6]
```diff
@@ -8,7 +8,11 @@
 */
 
 function studentPassed(grade) {
-
+  if (grade < 50) {
```
Comment (martynamaron) [inline week-2/G-conditionals-2/exercise-2.js]: nice!

File: week-2/F-logical-operators/exercise2.js [hunk 7]
```diff
@@ -5,17 +5,31 @@
...
-function isNegative() {}
-
+function isNegative(num) {
+  if (num < 0) {
+    return true;
+  } else {
+    return false;
+  }
+}
+function isBetween5and10(num) {
```
Comment (martynamaron) [inline week-2/F-logical-operators/exercise2.js]: You want `isBetween5and10`, `isLongName` and `startsWithD` **always** to return a boolean. What happens when you do `startsWithD("Zareen")`? How can you fix that?

File: week-2/F-logical-operators/exercise2.js [hunk 8]
```diff
@@ -5,17 +5,31 @@
...
+  if (num >= 5 && num <= 10) return true;
+}
+function isLongName(name) {
+  if (name.length > 3) return true;
+}
+function startsWithD(name) {
+  if (name.charAt(0) == "D") return true;
```
Comment (martynamaron) [inline week-2/F-logical-operators/exercise2.js]: Nice! But check what happens when you do `startsWithD("daniel")` - the string still starts with the letter `d`! :)

File: week-2/C-comparison-operators/exercise.js [hunk 9]
```diff
@@ -7,14 +7,14 @@
 
 var studentCount = 16;
 var mentorCount = 9;
-var moreStudentsThanMentors; // finish this statement
+var moreStudentsThanMentors = studentCount > mentorCount; // finish this statement
 
 var roomMaxCapacity = 25;
-var enoughSpaceInRoom; // finish this statement
+var enoughSpaceInRoom = roomMaxCapacity == 25; // finish this statement
```
Comment (martynamaron) [inline week-2/C-comparison-operators/exercise.js]: This variable should check if the combined total of students and mentors **is greater** than the room capacity. Please fix :)

**Comment by martynamaron** [comment]
Hey @khan-zareen ! Well done, nice work, I left some comments for improvement, please try and address them :) Also, don't forget the week-2-practice folder!

File: week-2/M-array-map/exercise.js (hunk 1 above)
Comment (khan-zareen) [reply week-2/M-array-map/exercise.js]: Thank you Martyna

File: week-2/I-array-properties/exercise.js (hunk 4 above)
Comment (khan-zareen) [reply week-2/I-array-properties/exercise.js]: we can simplify as follows: 
function isEmpty(arr) {
  return arr.size();
}

File: week-2/F-logical-operators/exercise2.js (hunk 7 above)
Comment (khan-zareen) [reply week-2/F-logical-operators/exercise2.js]: well i could just return the value and remove the if statement e.g.
function isBetween5and10(num) {
  if (num >= 5 && num <= 10) return true;
} 
could be changed to
function isBetween5and10(num) {
  return num >= 5 && num <= 10);
}

File: week-2/F-logical-operators/exercise2.js (hunk 8 above)
Comment (khan-zareen) [reply week-2/F-logical-operators/exercise2.js]: yes there should be strict equality there i.e. name.charAt(0) ==="D"

File: week-2/C-comparison-operators/exercise.js (hunk 9 above)
Comment (khan-zareen) [reply week-2/C-comparison-operators/exercise.js]: corrected as:
var enoughSpaceInRoom = roomMaxCapacity >= studentCount + mentorCount;

**Pull request closed without merging**